src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -82,7 +120,16 @@ public void updateOrderWithPoLines(String id, CompositePurchaseOrder compPO) {
       })
       .exceptionally(this::handleError);
 
-    return future;
+  }
+
+  private CompletableFuture<Void> updatePoLinesNumber(CompositePurchaseOrder compOrder, JsonArray poLinesFromStorage) {
+    List<CompletableFuture<?>> futures = poLinesFromStorage.stream().map(o -> {
+      JsonObject lineFromStorage = (JsonObject) o;
+      lineFromStorage.put(PO_LINE_NUMBER, buildNewPoLineNumber(lineFromStorage, compOrder.getPoNumber()));
+      String endpoint = String.format(URL_WITH_LANG_PARAM, resourceByIdPath(PO_LINES, lineFromStorage.getString(ID)), lang);
+      return operateOnSubObj(HttpMethod.PUT, endpoint, lineFromStorage, httpClient, ctx, okapiHeaders, logger);
```

Comment:
There are 2 methods to update PO line:
- handlePoLines - this method uses PutOrderLineByIdHelper to update PO lines
- updatePoLinesNumber - this method uses badly named operateOnSubObj static method to update PO lines
I would propose to introduce another method in PutOrderLineByIdHelper such as 'updateOrderLineSummary' that updates only PO line summary w/o sub-objects and use this method here.

Reply:
created updateOrderLineSummary method in PutOrderLineByIdHelper